### PR TITLE
Removed observers from a Model after finish of the stream

### PR DESCRIPTION
**Pull request by bigcup**
I removed flags and properly finished the stream. Destroy emits an end event for compatibility with ending of ReadableStream. I am not satisfied with redefinition of the destroy() method, because in node 8 all type of streams have a native destroy method, so I left default implementation of it.

#### Related issues

#1569 

### Checklist

- [x] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style
  guide](http://loopback.io/doc/en/contrib/style-guide.html)

/cc @ritch @bajtos can you please review the code?

### CODE REVIEW

**Comment by slnode** [comment]
Can one of the admins verify this patch? To accept patch and trigger a build add comment ".*ok\W+to\W+test.*"

**Comment by slnode** [comment]
Can one of the admins verify this patch?

**Comment by slnode** [comment]
Can one of the admins verify this patch?

**Comment by slnode** [comment]
Can one of the admins verify this patch?

**Comment by slnode** [comment]
Can one of the admins verify this patch?

**Comment by bajtos** [comment]
@slnode ok to test

File: test/change-stream.test.js [hunk 1]
```diff
@@ -67,6 +73,26 @@ describe('PersistedModel.createChangeStream()', function() {
         });
       });
     });
+
+    it('should not be able after destroy', function(done) {
```
Comment (superkhau) [inline test/change-stream.test.js]: please clear up this description

File: test/change-stream.test.js [hunk 2]
```diff
@@ -20,14 +21,19 @@ describe('PersistedModel.createChangeStream()', function() {
       });
     });
 
+    afterEach(function() {
+      var Score = this.Score;
+      expect(Score._observers['after save']).to.be.empty();
+      expect(Score._observers['after delete']).to.be.empty();
+    });
```
Comment (superkhau) [inline test/change-stream.test.js]: nit: extract to shared fn to end of describe block (ie. `afterEach(verifyScore)`) -- https://loopback.io/doc/en/contrib/style-guide.html

**Comment by bajtos** [comment]
Thank you for the new commit, @bigcup. Please leave also a comment next time, not everybody is receiving email notifications about new commits.

I squashed your two commits into a single one and improved the commit message to follow our [50/72 style](http://loopback.io/doc/en/contrib/git-commit-messages.html) - most notable changed the style to imperative.

Let's wait for CI results before landing.

BTW, the email address you have used to commit the changes is not linked to your GitHub profile (@bigcup). Could you please fix that? See
 - https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user
 - https://help.github.com/articles/adding-an-email-address-to-your-github-account/

**Comment by bajtos** [comment]
Landed, thank you for the contribution! 🎉

**Comment by bigcup** [comment]
@bajtos Thank you for review

**Comment by didikeke** [comment]
I'm using node v8.5.0 in server side and create an EventSource in Chrome to connect change-stream.  

However, the "removeHandlers" is never executed.  The require('stream').PassThrough doesn't emit any event (such as 'error', 'close', 'finish', 'end') even if I close/refresh the browser.

@bigcup Do you have any idea, how to let the removeHandlers be called? Thanks!

**Comment by bajtos** [comment]
>  I'm using node v8.5.0 in server side and create an EventSource in Chrome to connect change-stream.
>
> However, the "removeHandlers" is never executed. The require('stream').PassThrough doesn't emit any event (such as 'error', 'close', 'finish', 'end') even if I close/refresh the browser.
>
> Do you have any idea, how to let the removeHandlers be called? Thanks!

@didikeke hello, I am not very familiar with the low-level details of EventSource. However, I landed a fix to strong-remoting recently that can be related to the problem you are observing - see https://github.com/strongloop/strong-remoting/pull/417. Could you please check that your app is using the latest strong-remoting version (`3.4.1` at the time of this writing)? 

@perkd you are the author of that strong-remoting fix, could you perhaps advise here?

If the issue is still present even in the latest version, then I'd like to kindly ask you (@didikeke) to open a new GitHub issue and include a sample application that reproduces the problem (see http://loopback.io/doc/en/contrib/Reporting-issues.html#bug-report).

**Comment by didikeke** [comment]
@bajtos Thanks. It works after updating the lastest strong-remoting :)

**Comment by bajtos** [comment]
> Thanks. It works after updating the lastest strong-remoting :)

@didikeke that's great to hear 👍

**Pull request merged**